### PR TITLE
Create migration guide for ASP.NET applications

### PR DESCRIPTION
**Pull request by TimothyMothra**
This PR adds a guide for migrating ASP.NET applications to OpenTelemetry.
Our team decided to start with ASP.NET. 

### Notes
- We don't have a reputable example of configuring ILogger to ASP.NET applications.
- All the steps in this doc are "Step 1". This is to make my life easier when making large changes to this doc. We can fix this before publishing.

### CODE REVIEW

File: docs/migration/guide_aspnet.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Guide for Asp.Net applications
+
+## Intro
+
+The following guide details how to remove Application Insights from an ASP.NET web app and how to set up the OpenTelemetry SDK.
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:5]:
```suggestion
The following guide details how to migrate from Application Insights SDK to OpenTelemetry based solution for an ASP.NET web application.
```

File: docs/migration/guide_aspnet.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Prerequisites
+
+- An ASP.NET web application already instrumented with Application Insights
+- A actively supported version of .NET (link)
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:10]: Be specific as ASP.NET is only for .NET Framework, so .NET462 or higher only.

File: docs/migration/guide_aspnet.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Steps to Migrate
+
+### Step 1: Remove Application Insights SDK
+
+When you first added ApplicationInsights to your project, the SDK would have added a config file and made some edits to the web.config.
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:16]: https://learn.microsoft.com/en-us/azure/azure-monitor/app/asp-net#how-can-i-uninstall-the-sdk

File: docs/migration/guide_aspnet.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          <remove name="ApplicationInsightsWebTracking" />
+          <add name="ApplicationInsightsWebTracking" type="Microsoft.ApplicationInsights.Web.ApplicationInsightsHttpModule, Microsoft.AI.Web" preCondition="managedHandler" />
+        </modules>
+      </system.webServer>
+    </configuration>
+    ```
+
+### Step 1: Install the OpenTelemetry SDK and Enable at Application Startup
+
+The OpenTelemery SDK must be configured at application startup. This is typically done in the Global.asax.cs.
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:47]:
```suggestion
The OpenTelemery SDK must be configured at application startup. This is typically done in the `Global.asax.cs`.
```

File: docs/migration/guide_aspnet.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        this.meterProvider?.Dispose();
+    }
+}
+```
+
+For more examples, see the following guides:
+- OpenTelemetry SDK's getting started guide: https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry
+- ASP.NET example project: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/examples/AspNet/Global.asax.cs
+
+### Step 1: Configure Instrumentation Libraries.
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:86]: this is step2 right?

File: docs/migration/guide_aspnet.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            .Build();
+    }
+
+    protected void Application_End()
+    {
+        this.tracerProvider?.Dispose();
+        this.meterProvider?.Dispose();
+    }
+}
+```
```
Comment (cijothomas) [inline docs/migration/guide_aspnet.md:80]: i'd recommend what happens at the end of this step - in this case nothing happens, but make it very clear that this step is just a prep-step only, and no telemetry will be collected at this point.

File: docs/migration/guide_aspnet.md (hunk 5 above)
Comment (TimothyMothra) [reply docs/migration/guide_aspnet.md:86]: I'm saving this for last. Right now, all the Steps are listed as "1". I did this intentionally so I can focus on the content without needing to re-number every section when making edits.

**Comment by TimothyMothra** [comment]
Closing. will keep rough draft internal until ready to publish

**Pull request closed without merging**